### DISCUSSION TITLE
Set a trigger from Berry script

### DISCUSSION THREAD

**Discussion by msetina** [arendst/Tasmota#19592] [Q&A]
Is there a way to set a value to trigger a rule from Berry script?

**Comment by sfromis** [comment]
Yes, you can issue the `Event` command using `tasmota.cmd`. Anyway, I'd suggest doing it all in Berry, instead of mixing with rules.

**Comment by msetina** [comment]
I was also on the way to doing it in Berry, but would need to reimplement RuleTimer..
On the other hand I found that TriggerName is a name from a node in JSON. Does that mean that TriggerName would be nodes separated by #  that driver written in Berry would write with json_append() ?
I could not find any examples in that direction to confirm the words in documentation.

**Reply by sfromis** [reply]
Well, you *could* also create a driver in the style of sensor payloads using `json_append()`, and then use that for triggers. What I recommended was using the `Event` command which is very simple for "pushing" a trigger from a command, and that command can come from the console, MQTT, serial, http, or Berry. Did you look that command up?
You then can have a trigger like `on Event#shootme=bang do ... endon` with `Event shootme=bang`

When it comes to doing something like a RuleTimer, the simplest equivalent for that would be `tasmota.set_timer`. If your use case is something like repeats at intervals, the `cron` module can be great too.

**Reply by msetina** [reply]
I've tried to find Event command syntax to no avail. It probabaly is simple, but would be nice to see it in doc.

**Reply by sfromis** [reply]
https://tasmota.github.io/docs/Commands/#event links to https://tasmota.github.io/docs/Rules/#rule-trigger where one of the listed triggers is `Event#eventName`, and an example of how it works with values.

**Reply by msetina** [reply]
Yes. But this is for Rule. I do not find Event command syntax. Is it:
Event eventName value

**Reply by sfromis** [reply]
Read the link I posted.

> Example: Command `Event speed=2` in rule trigger `on event#speed` will have the `%value%` of 2.

Additionally, I already posted a similar example in this thread.

**Reply by msetina** [reply]
Thank you for your patience. I eventualy found it and I understand. It would be easier to see it in the Event command description.
If I may I would like to extend the basic discusion:
But there could be a difference if rules run on a different loop and allow sychronicity of gpio outputs turning on or off and berry would run turning outputs on or off sequentialy

**Reply by sfromis** [reply]
Anything with synchronicity would be a question about how your are implementing things, not something which can be judged on an abstract level. And "different loop" again needs to be specific to be part of considerations.
